s3parq/publish_parq.py

```diff
@@ -175,13 +180,14 @@ def _gen_parquet_to_s3(bucket: str, key: str, dataframe: pd.DataFrame,
 
     try:
         table = pa.Table.from_pandas(
-            df=dataframe, schema=_parquet_schema(dataframe), preserve_index=False)
+            df=dataframe, schema=_parquet_schema(dataframe, custom_redshift_columns=custom_redshift_columns), preserve_index=False)
     except pa.lib.ArrowTypeError:
         logger.warn(
             "Dataframe conversion to pyarrow table failed, checking object columns for mixed types")
         dataframe_dtypes = dataframe.dtypes.to_dict()
         object_columns = [
-            col for col, col_type in dataframe_dtypes.items() if col_type == "object"]
+            # Skip conversion to strings if the column contains decimal objects. Otherwise, decimal objects will be improperly converted.
```

Comment:
Can we add a section to the README about this support, and that at this time it only does special stuff for Decimal? Unless there's no other RedShift type we might handle in this way.
Same to the docstrings on the custom publish function.

Reply:
Thanks! Good call, I substantially improved the documentation for this.